### PR TITLE
feat: Add component logo to masthead

### LINKED ISSUE #53: Display component icon on service detail page
Show logo icon in component details page - either in masthead or on the About card.

image

### PR DESCRIPTION
**Pull request by SamoKopecky**
Resolves #53 

Add component logo to the masthead. This works by copying over the upstream source code for [`EntityLayout.tsx`](https://github.com/backstage/backstage/blob/master/plugins/catalog/src/components/EntityLayout/EntityLayout.tsx) and its dependencies. Then the code is edited to add the component logo. See the image below.

Also add a [constants file](https://github.com/operate-first/service-catalog/pull/87/commits/ccd9253c095f7e0c20b6dd14181d6ebea5cbab00) for `ICON_ANNOTATION`.

image

You can check the last commit to see how the logo is implemented, the first commit is just copy pasting stuff from backstage upstream.

### CODE REVIEW

File: packages/app/src/App.tsx [hunk 1]
```diff
@@ -108,3 +108,4 @@ const App = () => (
 );
 
 export default App;
+export const ICON_ANNOTATION = 'operate-first.cloud/logo-url';
```
Comment (tumido) [inline packages/app/src/App.tsx:111]: This is introducing circular dependency... let's not do that. If necessary create a separate file for constants...

**Review by tumido — CHANGES_REQUESTED** [review]

File: packages/app/src/App.tsx (hunk 1 above)
Comment (SamoKopecky) [reply packages/app/src/App.tsx]: Resolved in the latest commit.

**Comment by tumido** [comment]
I'm really sorry here. While I really like the outcome, I don't like the implementation. The huge copypaste from upstream is not maintainable... isn't there a better/easier way?

**Comment by SamoKopecky** [comment]
> I'm really sorry here. While I really like the outcome, I don't like the implementation. The huge copypaste from upstream is not maintainable... isn't there a better/easier way?

Yea I very much agree I tried finding different solutions but I couldn't find any. I'll search around more for a better solution.

**Comment by SamoKopecky** [comment]
/close 

Since the UI got a big rework, this doesn't make sense anymore.

**Pull request closed without merging**